src/dmd_era5/era5_download/era5_download.py

```diff
@@ -128,19 +134,91 @@ def download_era5_data(parsed_config: dict, use_mock_data: bool = False) -> xr.D
         raise ValueError(msg) from e
 
 
-def main(use_mock_data: bool = False) -> None:
-    """Main function to run the ERA5 download process."""
+def main(
+    config: dict = config, use_mock_data: bool = False, use_dvc: bool = False
+) -> tuple[bool, bool]:
+    """
+    Main function to run the ERA5 download process.
+    If using DVC, the function will attempt to retrieve the data from DVC first
+    before downloading it.
+
+    Args:
+        config (dict): Configuration dictionary with the configuration parameters,
+        optional and primarily for testing purposes.
+        use_mock_data (bool): Whether to use mock data instead of downloading,
+        for testing purposes.
+        use_dvc (bool): Whether to use Data Version Control (DVC) to track the data.
+
+    Returns:
+        tuple[bool, bool]: A tuple of two booleans indicating whether the data was
+        added to DVC and whether it was retrieved from DVC.
+    """
+
+    added_to_dvc = False
+    retrieved_from_dvc = False
     try:
-        parsed_config = config_parser(
-            config=config, section="era5-download", logger=logger
-        )
-        download_era5_data(parsed_config, use_mock_data)
-        log_and_print(logger, "ERA5 download process completed successfully.")
+        parsed_config = config_parser(config, section="era5-download", logger=logger)
+
+        def handle_download_and_dvc() -> bool:
+            """Helper function to download data and add it to DVC."""
+            era5_ds = download_era5_data(parsed_config, use_mock_data)
+            log_and_print(logger, "ERA5 download process completed successfully.")
+            if use_dvc:
+                try:
+                    log_and_print(logger, "Adding ERA5 slice to DVC...")
+                    add_data_to_dvc(parsed_config["save_path"], era5_ds.attrs)
+                    log_and_print(logger, "ERA5 slice added to DVC.")
+                    return True
+                except Exception as e:
+                    log_and_print(
+                        logger, f"Error adding ERA5 slice to DVC: {e}", level="error"
+                    )
+                    return False
+            return False
+
+        if use_dvc:
+            log_and_print(logger, "Attempting to retrieve ERA5 slice from DVC...")
+            try:
+                retrieve_data_from_dvc(parsed_config)
+                log_and_print(
+                    logger,
+                    f"ERA5 slice retrieved from DVC: {parsed_config['save_path']}",
+                )
+                retrieved_from_dvc = True
+            except (FileNotFoundError, ValueError) as e:
+                log_and_print(logger, f"Could not retrieve ERA5 slice from DVC: {e}")
+                added_to_dvc = handle_download_and_dvc()
+        else:
+            added_to_dvc = handle_download_and_dvc()
```

Comment:
There are quite a few branches and exception handling statements here, could these be simplified? For example, at the moment, the `handle_download_and_dvc` helper is called in separate branches for True and False values of `use_dvc`, but the helper function immediately checks this value again and branches.

Reply:
Have edited so that `use_dvc` is only checked once